docs/import/create_score_asset.js

```diff
@@ -319,47 +322,86 @@ function createScoreAsset(
+      .catch((renameErr) => {
+        console.log(renameErr);
+        // These checks aren't perfect detection, but best we can do.
+        if (renameErr.includes('does not exist')) {
+          console.log('Old ' + scoreAssetPath + ' not found, did not move it');
+        } else if (renameErr.includes('Cannot overwrite asset')) {
+          // Delete and try again.
+          return new Promise(
+              (deleteResolve, deleteReject) =>
+                  ee.data.deleteAsset(oldScoreAssetPath, (_, deleteErr) => {
+                    if (deleteErr) {
+                      // Don't try to recover here.
+                      // Don't show deletion error to user, but log it
+                      // somewhere.
+                      console.error(deleteErr);
+                      const message =
+                          'Error moving old score asset: ' + renameErr;
+                      setStatus(message);
+                      deleteReject(message);
+                    } else {
+                      // Delete succeeded, try again.
+                      deleteResolve(renameAssetAsPromise(
+                          scoreAssetPath, oldScoreAssetPath));
+                    }
+                  }));
         } else {
-          const message = 'Error deleting: ' + err;
+          const message = 'Error moving old score asset: ' + renameErr;
           setStatus(message);
-          reject(new Error(message));
-          return;
-        }
-      }
-      ee.data.renameAsset(scoreAssetPath, oldScoreAssetPath, (_, err) => {
-        if (err) {
-          if (err.includes('does not exist')) {
-            // This check isn't perfect detection, but best we can do.
-            console.log(
-                'Old ' + scoreAssetPath + ' not found, did not move it');
-          } else {
-            const message = 'Error backing up: ' + err;
-            setStatus(message);
-            reject(new Error(message));
-            return;
-          }
+          console.error(message);
+          throw renameErr;
         }
+      })
+      .then(() => {
         task.start();
         $('#upload-status')
             .text(
                 'Check Code Editor console for upload progress. Task: ' +
                 task.id);
-        resolve(task);
+        return task;
       });
-    });
-  });
+}
+
+/**
+ * Converts {@link ee.data.renameAsset} into a Promise, by resolving/rejecting
+ * in its callback.
+ * @param {string} from See {@link ee.data.renameAsset}
+ * @param {string} to See {@link ee.data.renameAsset}
+ * @return {Promise<?Object>} Promise that resolves if rename is successful,
+ *     and completes with rejection if there is an error. The rejection is
+ *     just a string, not an `Error` because that is what EarthEngine does. The
+ *     resolve will typically be with an Object that has information about the
+ *     renamed asset, but EarthEngine doesn't guarantee that
```

Comment:
eurgh, any idea what it might be if not that object?

Reply:
EE docs say it can be null. But doesn't matter, the returned value is irrelevant, because we know what the asset's name is going to be. We only use it for the error/non-error signal.